core/src/main/java/org/javarosa/core/model/QuestionDef.java

```diff
@@ -48,7 +49,7 @@
  *
  * @author Daniel Kayiwa/Drew Roos
  */
-public class QuestionDef implements IFormElement, Localizable {
+public class QuestionDef extends ActionController implements IFormElement, Localizable {
```

Comment:
Yea, it's pretty abstracted away.. In `XFormPasrser.parseControlChildren()` if `actionHandlers` contains a handler for the element, it calls handle on the element, with its parent `QuestionDef`as the `parent` argument to `handle()`. If you follow that all the way down the chain of `parseAction()` --> `parseSetValueAction()` that `parent` object from the original call to `handle()` is passed to `parseSetValueAction()` as the `source` argument, and calls `source.registerEventListener(event, action)`.